### PR TITLE
Auto lock on speaker when getting ready to shoot

### PR DESCRIPTION
**Pull request by Tyson453**

### CODE REVIEW

File: robotbase/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java [hunk 1]
```diff
@@ -118,4 +122,10 @@ public void accept(ChassisSpeeds speeds) {
       }
     };
   }
+
+  public double getSpeakerLockRotation() {
+    double tx = Robot.shooterLimelight.getTX();
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java:127]: You may want to check if there is a valid target, or make sure the tx returns zero when it does not see a target

File: robotbase/src/main/java/frc/robot/controls/DriverControls.java [hunk 2]
```diff
@@ -16,19 +19,28 @@ public class DriverControls implements RumbleInterface {
...
+    m_rightTriggerShoot = new AxisThresholdTrigger(m_controller, Axis.kRightTrigger, .6);
+
     mapControls();
   }
 
   public void mapControls() {
     m_backButton.onTrue(new InstantCommand(() -> Robot.drive.setYaw(0)));
     m_startButton.onTrue(new InstantCommand(() -> Robot.drive.setYaw(180)));
+
+    m_rightTriggerPrime.whileTrue(new InstantCommand(() -> Robot.state.setDriveControlState(DriveControlState.SPEAKER_LOCK)));
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/controls/DriverControls.java:42]: Do you need to set the limelight pipeline here?

**Review by NolanCampbell2357 — CHANGES_REQUESTED** [review]

File: robotbase/src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -76,10 +76,10 @@ public void robotInit() {
         new DriverControls(
             new XboxController(Constants.CONTROLLER.DRIVE_CONTROLLER_PORT),
             Constants.CONTROLLER.DRIVE_CONTROLLER_DEADBAND);
-    codriverControls =
-        new CodriverControls(
-            new XboxController(Constants.CONTROLLER.CODRIVER_CONTROLLER_PORT),
-            Constants.CONTROLLER.CODRIVE_CONTROLLER_DEADBAND);
+    // codriverControls =
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/Robot.java:79]: Can we uncomment this?

File: robotbase/src/main/java/frc/robot/commands/EnableSpeakerLockCommand.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Robot;
+import frc.robot.state.RobotState.DriveControlState;
+
+public class EnableSpeakerLockCommand extends Command {
+  @Override
+  public void initialize() {
+    Robot.state.setDriveControlState(DriveControlState.SPEAKER_LOCK);
+    Robot.shooterLimelight.setPipeline(4);
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/commands/EnableSpeakerLockCommand.java:11]: We'll need to add an enum, or better way to switch pipelines other than hardcoding a number.

File: robotbase/src/main/java/frc/robot/commands/EnableSpeakerLockCommand.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  }
+
+  // @Override
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/commands/EnableSpeakerLockCommand.java:14]: Remove unused code

File: robotbase/src/main/java/frc/robot/util/PipelineManager.java [hunk 6]
```diff
@@ -1,18 +1,16 @@
...
-      case Red:
-        return Constants.SHOOTER_LIMELIGHT.RED_SPEAKER_APRILTAG_PIPELINE_INDEX;
-      case Blue:
-        return Constants.SHOOTER_LIMELIGHT.BLUE_SPEAKER_APRILTAG_PIPELINE_INDEX;
-      default:
-        System.err.println("----- Alliance not selected -----");
-        return 0;
-    }
+    return 4;
+    // switch (Robot.state.getAlliance()) {
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/util/PipelineManager.java:6]: Fix this or remove it

**Review by NolanCampbell2357 — CHANGES_REQUESTED** [review]
Remember to rebase off main

File: robotbase/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java [hunk 7]
```diff
@@ -128,10 +136,32 @@ public Consumer<ChassisSpeeds> getChassisSpeedsConsumer() {
...
       }
     };
   }
+
+  public ChassisSpeeds getChassisSpeeds() {
+    ChassisSpeeds chassisSpeeds = getKinematics().toChassisSpeeds(getModuleStates());
+    return chassisSpeeds;
+  }
+
+  public double getTargetLockRotations() {
```
Comment (NolanCampbell2357) [inline robotbase/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java:151]: Looks like you are not calling this anymore. It looks like you had some merge issues.

**Review by NolanCampbell2357 — CHANGES_REQUESTED** [review]

File: robotbase/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java (hunk 7 above)
Comment (Tyson453) [reply robotbase/src/main/java/frc/robot/subsystems/CommandSwerveDrivetrain.java]: Fixed

**Review by NolanCampbell2357 — APPROVED** [review]

**Pull request merged**